symmer/operators/noncontextual_op.py

```diff
@@ -313,22 +313,42 @@ def noncontextual_generators(self) -> None:
         """ 
         Find an independent generating set for the noncontextual operator.
         """
-        # identify the symmetry generating set
-        self.symmetry_generators = IndependentOp.symmetry_generators(self, commuting_override=True)
-        # mask the symmetry terms within the noncontextual operator
-        _, symmetry_mask = self.generator_reconstruction(self.symmetry_generators)
-        # identify the reamining commuting cliques
-        self.decomposed = self[~symmetry_mask].clique_cover(edge_relation='C')
+
+        ## rather than searching over self (operator iteself) use generators instead ==> much faster
+        # get Z2 symmetry of generators (may anticommute among themselves, but will commute with all generators)
+        Z2_symmerties = IndependentOp.symmetry_generators(self.generators, commuting_override=True)
```

Comment:
faster way of finding Z2 syms (note self.generators is cached and aleady calculated in noncon check)